datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestStickyPartitionAssignment.java

```diff
@@ -87,11 +86,24 @@ public void testAddPartitions() {
     DatastreamGroupPartitionsMetadata newPartitionsMetadata =
         new DatastreamGroupPartitionsMetadata(datastreams.get(0), newPartitions);
 
-    assignment = strategy.assignPartitions(assignment, newPartitionsMetadata);
+    Map<String, Set<DatastreamTask>> newAssignment = strategy.assignPartitions(assignment, newPartitionsMetadata);
 
-    for (DatastreamTask task : assignment.get("instance1")) {
+    for (DatastreamTask task : newAssignment.get("instance0")) {
       Assert.assertEquals(task.getPartitionsV2().size(), 2);
     }
+
+    Map<String, List<DatastreamTask>> taskToCleanup = strategy.getTasksToCleanUp(datastreams, newAssignment);
+    Assert.assertEquals(taskToCleanup.size(), 0);
+
+    // Adding the dependency task as well in the assignment list to simulate the scenario where
+    // the dependency task nodes are not deleted and the leader gets interrupted, OOM or hit session expiry.
+    // The next leader should be able to identify and cleanup.
+    Map<String, Set<DatastreamTask>> finalAssignment = assignment;
+    newAssignment.forEach((instance, taskSet1) -> taskSet1.addAll(finalAssignment.get(instance)));
+
+    taskToCleanup = strategy.getTasksToCleanUp(datastreams, newAssignment);
+    Assert.assertEquals(taskToCleanup.size(), 1);
+    taskToCleanup.forEach((instance, taskList1) -> Assert.assertEquals(taskList1.size(), 3));
```

Comment:
Should we also validate that the tasksToCleanup only contains the tasks we expect it to by matching task names?